application/common/models/User.php

```diff
@@ -636,13 +634,20 @@ public function beforeSave($insert): bool
     /**
      * @param string $inviteCode
      * @return User|null
+     * @throws \Exception
      * @throws NotFoundException
-     * @throws NotFoundHttpException
+     * @throws \Sil\Idp\IdBroker\Client\ServiceException
      */
     public static function getUserFromInviteCode(string $inviteCode)
     {
-        /** @var IdBrokerClient $client */
-        $client = \Yii::$app->passwordStore->getClient();
+        /** @var IdBroker $passwordStore */
+        $passwordStore = \Yii::$app->passwordStore;
+
+        if (! (method_exists($passwordStore, 'getClient'))) {
```

Comment:
It may very well be overkill, but do you think it would be worth defining an interface for Password Stores that support the new user invite? That could make this if-condition more readable (`if ($passwordStore implements NewUserInviteInterface) { ... }`) rather than having to know that the presence of a `getClient()` method means that it supports new user invites.